docs/LexerDevelopment.md

```diff
@@ -431,18 +465,43 @@ returns true should be tested.
 The demo file is tested automatically as part of Rouge's test suite. The file
 should be able to be parsed without producing any `Error` tokens.
 
+The demo is also used on [rouge.jneen.net][hp] as the default text to display
+when a lexer is chosen. It should be short (less than 20 lines if possible).
+
+[hp]: http://rouge.jneen.net/
+
+Your demo must be saved to `lib/rouge/demos/<name_of_your_lexer>`. Please note
+that there is no file extension.
+
 ### Visual Samples
 
-While the visual sample is tested by the testing suite to ensure that it does
-not raise any errors, the primary purpose is to allow the user to quickly scan
-through a large sample of text in a particular language and make sure that the
-highlighting looks correct.
+A visual sample is a file that includes a representive sample of the syntax of
+your language. The sample should be long enough to reasonably demonstrate the
+correct lexing of the language but does not need to offer complete coverage.
+While it can be tempting to copy and paste code found online, please refrain
+from doing this. If you need to copy code, indicate in a comment (using the
+appropriate syntax for your lexer's language) the source of the code.  Avoid
+including code that is duplicative of the other code in the sample.
 
 If you are adding or fixing rules in the lexer, please add some examples of the
 expressions that will be highlighted differently to the visual sample if
 they're not already present. This greatly assists in reviewing your lexer
 submission.
 
+Your visual sample must be saved to `spec/visual/sample/<name_of_your_lexer>`.
+As with the demo file, there is no file extension.
+
+### Running the Tests
+
+The spec and the demo can be run using the `rake` command. You can run this by
+typing `bundle exec rake` at the command line. If everything works, you should
+see a series of dots. If you have an error, this will appear here, too.
+
+To see your visual sample, launch Rouge's visual test app by running `bundle
+exec rackup`. You can choose your sample from the complete list by going to
+<http://localhost:9292>.  ot have a file extension. To start the test suite, run
```

Comment:
Great document ! It looks like there is a missing piece of thee sentence here. I'd be happy to send a pull request to fix this but I don't know what you wanted to write.

Reply:
Hahahaha, you've got me—I'm not sure! I _think_ this is an errant copy and paste and everything after 'ot' in this paragraph should be deleted.